### ISSUE TITLE
parallel 2.0.0 has been released, but rubocop's gemspec is limiting to ~>1.10, preventing an update.

### ISSUE THREAD

**Issue by jmaino** [rubocop/rubocop#15081]
### Before submitting

- [x] #15082
- [x] I am running the latest stable version of RuboCop.

### Issue type

False positive

### Steps to reproduce the problem

Usin rubocop 1.86.0, gemspec limits parallel to (~> 1.10), preventing update/upgrade to parallel 2.0.0

### Expected behavior

If there are no breaking changes, please permit use of newest version of parallel

### Actual behavior

Update of gems to newest parallel prohibited

### RuboCop configuration

```yaml

```

### RuboCop version

```shell
1.86.0
```
